src/pages/MainPage.jsx

```diff
@@ -9,75 +9,105 @@ import Sidebar from '../components/dashboard/Sidebar.jsx';
-  'https://images.unsplash.com/photo-1617330070571-60434437a26c?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1867&q=80',
-  'https://images.unsplash.com/photo-1617504546842-5768342f1c46?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2102&q=80',
-  'https://images.unsplash.com/photo-1617510031576-c9db88171057?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2102&q=80',
-  'https://images.unsplash.com/photo-1617505907947-9cb31eb80183?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1867&q=80',
-  'https://images.unsplash.com/photo-1617505481649-8d0accca73d6?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2100&q=80',
-  'https://images.unsplash.com/photo-1617330070571-60434437a26c?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1867&q=80',
-  'https://images.unsplash.com/photo-1617504546842-5768342f1c46?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2102&q=80',
-  'https://images.unsplash.com/photo-1617510031576-c9db88171057?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=2102&q=80',
-  'https://images.unsplash.com/photo-1617505907947-9cb31eb80183?ixid=MXwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHw%3D&ixlib=rb-1.2.1&auto=format&fit=crop&w=1867&q=80',
-];
+import Groups from '../models/groups'; // Dummy group list.
+import GroupsStateContext from '../contexts/GroupStateContext.jsx';
 
 function MainPage() {
-  const [user, setUser] = useState(null);
+  const [user, setUser] = useState({});
+  const [groupData, dispatch] = useReducer(groupReducer, { groups: [], index: 0 });
+  const { groups, index } = groupData;
+  const [photos, setPhotos] = useState([]);
+  const [groupTitle, setGroupTitle] = useState('');
   const history = useHistory();
 
+  function buildPhotoList() {
+    if (groups !== undefined && groups.length < 1) {
+      setPhotos([]);
+    } else if (groups !== undefined) {
+      const { images } = groupData.groups[groupData.index];
+      setGroupTitle(groups[index].title);
+      setPhotos(images.map(img => img.URL));
+    }
+  }
+
   async function getUser(token, id) {
     try {
       API.getInfo(token, id)
         .then((res) => {
           setUser(res);
+          // Leaving this for now. Later will need to set group to the group
+          // list returned.
+          // dispatch({ type: 'init', state: { groups: Groups } });
         });
     } catch (e) {
       // TODO: Will probably need better error handling
       history.replace('/');
     }
   }
 
+  // Only want this to trigger once to grab user token and id.
   useEffect(() => {
     const token = localStorage.getItem('token');
     const id = localStorage.getItem('id');
     getUser(token, id);
```

Comment:
Since `getUser` is async, there should be an `await`, that way we don't try to keep loading the main page if `getUser` fails